index-es6.js

```diff
@@ -0,0 +1,693 @@
+   * ensure to call `resized()` method.
+   * @param {function(number)} callback
+   */
+  onResizeRequest(callback) {
+    this.onResizeRequest_ = callback;
+    Promise.resolve().then(() => {
+      if (this.requestedHeight_ != null) {
+        callback(this.requestedHeight_);
+      }
+    });
+  }
+
+  /**
+   * Signals back to the activity implementation that the client has updated
+   * the activity's size.
+   */
+  resized() {
+    if (!this.connected_) {
+      return;
+    }
+    const height = this.iframe_.offsetHeight;
+    this.messenger_.sendCommand('resized', {'height': height});
+  }
+
+  /**
+   * @param {string} cmd
+   * @param {?Object} payload
+   * @private
+   */
+  handleCommand_(cmd, payload) {
+    if (cmd == 'connect') {
+      // First ever message. Indicates that the receiver is listening.
+      this.connected_ = true;
+      this.messenger_.sendCommand('start', this.args_);
+      this.connectedResolver_();
+    } else if (cmd == 'result') {
+      // The last message. Indicates that the result has been received.
+      if (this.resultResolver_) {
+        const code = /** @type {!ActivityResultCode} */ (payload['code']);
+        const data =
+            code == ActivityResultCode.FAILED ?
+            new Error(payload['data'] || '') :
+            payload['data'];
+        const result = new ActivityResult(code, data);
+        this.resultResolver_(result);
+        this.resultResolver_ = null;
+        this.messenger_.sendCommand('close');
+        this.disconnect();
+      }
+    } else if (cmd == 'ready') {
+      if (this.readyResolver_) {
+        this.readyResolver_();
+        this.readyResolver_ = null;
+      }
+    } else if (cmd == 'resize') {
+      this.requestedHeight_ = /** @type {number} */ (payload['height']);
+      if (this.onResizeRequest_) {
+        this.onResizeRequest_(this.requestedHeight_);
+      }
+    }
+  }
+}
+
+
+
+/**
+ * The page-level activities manager. This class is intended to be used as a
+ * singleton. It can start activities as well as implement them.
+ */
+class Activities {
+
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @private @const {!Window} */
+    this.win_ = win;
+  }
+
+  /**
+   * Start an activity within the specified iframe.
+   * @param {!HTMLIFrameElement} iframe
+   * @param {string} url
+   * @param {string} origin
+   * @param {?Object=} opt_args
+   * @return {!Promise<!ActivityIframePort>}
+   */
+  openIframe(iframe, url, origin, opt_args) {
+    const port = new ActivityIframePort(iframe, url, origin, opt_args);
+    return port.connect().then(() => port);
+  }
+
+  /**
+   * Start activity implementation handler (host).
+   * @return {!Promise<!ActivityHost>}
+   */
+  connectHost() {
+    const host = new ActivityIframeHost(this.win_);
+    return host.connect().then(() => host);
+  }
+}
+
+
+
+/**
+ * This file is used to publish to NPM (https://www.npmjs.com/package/web-activities).
```

Comment:
This should go up top as `@fileoverview`.

Reply:
Done